app/presenters/bid_status_flash_factory.rb

```diff
@@ -7,8 +7,10 @@ def initialize(auction:, user:)
   end
 
   def create
-    if available?
+    if available? && auction.type == 'reverse'
```

Comment:
In this case, it is the `Factory` that is querying the type and figuring out which presenter to return. We could make the factory a model and have it return the presenters -- does that feel more intuitive to you?